src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/base_js.html.twig

```diff
@@ -46,22 +49,36 @@
                 xhr.send(payload || '');
             },
 
-            hasClass = function(el, klass) {
-                return el.className && el.className.match(new RegExp('\\b' + klass + '\\b'));
+            hasClass = function(el, cssClass) {
+                if (el.classList) {
+                    return el.classList.contains(cssClass);
+                } else if (el.className) {
+                    return el.className.match(new RegExp('\\b' + cssClass + '\\b'));
+                }
             },
 
-            removeClass = function(el, klass) {
-                if (el.className) {
-                    el.className = el.className.replace(new RegExp('\\b' + klass + '\\b'), ' ');
+            removeClass = function(el, cssClass) {
+                if (el.classList) {
+                    el.classList.remove(cssClass);
+                } else if (el.className) {
+                    el.className = el.className.replace(new RegExp('\\b' + cssClass + '\\b'), ' ');
                 }
             },
 
-            addClass = function(el, klass) {
-                if (!hasClass(el, klass)) {
-                    el.className += " " + klass;
+            addClass = function(el, cssClass) {
+                if (!hasClass(el, cssClass)) {
+                    if (el.classList) {
+                        el.classList.add(cssClass);
+                    } else if (el.className) {
+                        el.className += " " + cssClass;
+                    }
                 }
             },
 
+            toggleClass = function(el, cssClass) {
+                hasClass(el, cssClass) ? removeClass(el, cssClass) : addClass(el, cssClass);
```

Comment:
This is the current code:

```js
toggleClass = function(el, cssClass) {
    hasClass(el, cssClass) ? removeClass(el, cssClass) : addClass(el, cssClass);
}
```

And if I understand you right, you are asking me to do this:

```js
toggleClass = function(el, cssClass) {
    if (el.classList) {
        el.classList.toggle(cssClass);
    } else {
        hasClass(el, cssClass) ?  removeClass(el, cssClass) } : addClass(el, cssClass);
    }
}
```

Maybe I'm missing something :confused: